### PR TITLE
Add guide for using Azure Marketplace

### PR DESCRIPTION
**Pull request by kota2and3kan**
This PR add the guide for using Azure Marketplace.

I created new document (`docs/AzureMarketplaceGuide.md`).
And I added the recommendation to use Azure Marketplace (Azure Container Registry) from AKS in the `docs/ManualDeploymentGuideScalarDLOnAzure.md`.

Please take a look!

### CODE REVIEW

**Review by ymorimo — COMMENTED** [review]
I confirmed the deployment was successful with the marketplace plans. I left some comments. Please take a look!

File: docs/AzureMarketplaceGuide.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+1. Create a `reg-acr-secrets` secret resource for pulling Scalar product images from your private container registry.
+   ```console
+   kubectl create secret docker-registry reg-acr-secrets \
+     --docker-server=<your private container registry login server> \
+     --docker-username=<Service principal ID> \
+     --docker-password=<Service principal password>
+   ```
+
+1. Update a custom values file of Helm Charts.
```
Comment (feeblefakie) [inline docs/AzureMarketplaceGuide.md:70]:
```suggestion
1. Update the custom values file of the Helm Chart of a Scalar product you want to install.
```

File: docs/AzureMarketplaceGuide.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+             ```yaml
+             schemaLoading:
+               image:
+                 repository: "example.azurecr.io/scalarinc/scalardl-schema-loader"
+                 version: "3.4.0"
+               imagePullSecrets:
+                 - name: "reg-acr-secrets"
+             ```
+
+1. Deploy Scalar products using Helm Charts with the above custom values file.
```
Comment (feeblefakie) [inline docs/AzureMarketplaceGuide.md:140]:
```suggestion
1. Deploy the Scalar product using the Helm Chart with the above custom values file.
```

File: docs/AzureMarketplaceGuide.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+         helm install scalardl-auditor scalar-labs/scalardl-audit -f ./scalardl-auditor-custom-values.yaml
+         ```
+       * Scalar DL Schema Loader
+         ```console
+         helm install schema-loader scalar-labs/schema-loading -f ./schema-loader-custom-values.yaml
+         ```
+
+## Deploy containers on AKS (Azure Kubernetes Service) from your private container registry using Scalar Helm Charts
+
+If you deploy containers on the AKS (Azure Kubernetes Service), you don't need to create a `service principal` and `reg-acr-secrets`.  Your private container registry (Azure Container Registry) can allow access from your AKS.
```
Comment (feeblefakie) [inline docs/AzureMarketplaceGuide.md:161]:
```suggestion
If you deploy containers on AKS (Azure Kubernetes Service), you don't need to create a `service principal` and `reg-acr-secrets`.  Your private container registry (Azure Container Registry) allows access from your AKS.
```

File: docs/AzureMarketplaceGuide.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+1. Specify your private container registry (Azure Container Registry) when you create the AKS cluster.
```
Comment (feeblefakie) [inline docs/AzureMarketplaceGuide.md:163]:
```suggestion
1. Specify your private container registry (Azure Container Registry) when you create an AKS cluster.
```

File: docs/AzureMarketplaceGuide.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+   * GUI (Azure Portal)
+      * At the `Azure Container Registry` parameter in the `Integrations` tab, please specify your private container registry.
+   * CLI ([az aks create](https://docs.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#az-aks-create) command)
+      * Please specify `--attach-acr` flag with the name of your private container registry.
+   * Note: 
+      * Also, you can configure Azure Container Registry integration for existing AKS clusters using [az aks update](https://docs.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#az-aks-update) command with `--attach-acr` flag.
+      * Please refer to the [Azure Official Document](https://docs.microsoft.com/en-us/azure/aks/cluster-container-registry-integration) for more details.
+
+1. Update a custom values file of Helm Charts.
```
Comment (feeblefakie) [inline docs/AzureMarketplaceGuide.md:172]:
```suggestion
1. Update the custom values file of the Helm Chart of a Scalar product you want to install.
```

File: docs/AzureMarketplaceGuide.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+   * You need to specify your private container registry and the version (tag) as the value of `[].image.repository` and `[].image.version (tag)` in the custom values file.
+   * You do NOT need to specify the `reg-acr-secrets` as the value of `[].imagePullSecrets`, because your private container registry (Azure Container Registry) allows access from your AKS nodes.
+   * Examples
+      * Please refer to the `Deploy containers on Kubernetes other than AKS (Azure Kubernetes Service) from your private container registry using Scalar Helm Charts` section of this document.
+
+1. Deploy Scalar products using Helm Charts with the above custom values file.
```
Comment (feeblefakie) [inline docs/AzureMarketplaceGuide.md:178]:
```suggestion
1. Deploy the Scalar product using the Helm Chart with the above custom values file.
```

File: docs/ManualDeploymentGuideScalarDLOnAzure.md [hunk 7]
```diff
@@ -83,6 +83,7 @@ Install the following tools on the bastion for controlling the AKS cluster:
 * You should use Kubernetes node size `Standard D2s v3` for Scalar DL node pool.
 * You should create 3 nodes in each node group for high availability in production.
 * You should configure `Autoscale` for the node pools If you want to scale the nodes.
+* You should allow AKS cluster to pull containers from your private container registry (Azure Container Registry) if you use Azure Marketplace to get Scalar product containers. Please refer to [this guide](./AzureMarketplaceGuide.md) for more details of the Azure Marketplace.
```
Comment (feeblefakie) [inline docs/ManualDeploymentGuideScalarDLOnAzure.md:86]:
```suggestion
* You should allow AKS cluster to pull containers from your private container registry (Azure Container Registry) if you use Azure Marketplace to get Scalar products. Please refer to [this guide](./AzureMarketplaceGuide.md) for more details of the Azure Marketplace.
```

**Review by feeblefakie — COMMENTED** [review]
Very sorry for the late review.
Overall looking good!
Left some comments and suggestions, mostly for English correction.
PTAL!

File: docs/AzureMarketplaceGuide.md [hunk 8]
```diff
@@ -0,0 +1,178 @@
+# How to use Scalar product containers from Microsoft Azure Marketplace
+
+Scalar products (Scalar DB, Scalar DL, and some tools) are provided in the Microsoft Azure Marketplace as container images. This guide explains how to use Scalar product container images from the Microsoft Azure Marketplace.
```
[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/AzureMarketplaceGuide.md:3]: I think I remained this whitespace according to the markdown syntax to add new line.
However it is not necessary in the GitHub. I will take care of it. Thank you.

File: docs/AzureMarketplaceGuide.md [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+1. Sign in to the Microsoft Azure Marketplace using your work email address
+   * Please use the work email address that is used as an account of Microsoft Azure.
+        * In the after steps, we will create a private container registry (Azure Container Registry).
+   * If you have already signed in, this step will be skipped automatically.
+
+1. Input your information.
+   * `Company` is not required, but please input it.
+
+1. Select a `Plan` you need from the pull-down.
+   * `Plan` means a combination of the container image and the license. Please select the `Plan` you use.
```
[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/AzureMarketplaceGuide.md]: Yes. "Software plan" is correct. Thank you.
I fixed in 789fe24d08352fa2d393be9d1e17927ecebe788a.

File: docs/AzureMarketplaceGuide.md [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+* Note:
+    * Some Scalar products are licensed under a commercial license, and the Microsoft Azure Marketplace provides them as BYOL (Bring Your Own License). Please make sure you have the appropriate license.
+
+## Get Scalar products from Microsoft Azure Marketplace
+
+1. Access to the Microsoft Azure Marketplace.
+   * [Scalar DB](https://azuremarketplace.microsoft.com/en/marketplace/apps/scalarinc.scalardb)
+   * [Scalar DL](https://azuremarketplace.microsoft.com/en/marketplace/apps/scalarinc.scalardl)
+
+1. Select `Get It Now`.
```
[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/AzureMarketplaceGuide.md]: I used backticks for emphasizing without paying attention to the difference between backticks and asterisks. Thank you.
I fixed to use backticks for emphasizing
(I use backticks only command, flag, and parameter name.)

File: docs/AzureMarketplaceGuide.md [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+
+1. Sign in to the Microsoft Azure Marketplace using your work email address
+   * Please use the work email address that is used as an account of Microsoft Azure.
+        * In the after steps, we will create a private container registry (Azure Container Registry).
```
[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/AzureMarketplaceGuide.md]: Yes. This is a redundancy sentence. Thank you.
I removed this sentence in f806fe7b7b048ef203da7cf9d2e0f757bc6e3f7b.

File: docs/AzureMarketplaceGuide.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (ymorimo) [reply docs/AzureMarketplaceGuide.md]: A single `*` renders words *italic*, is it okay? To make them be **bold**, we need to quote them with `**`.

File: docs/AzureMarketplaceGuide.md [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+## Get Scalar products from Microsoft Azure Marketplace
+
+1. Access to the Microsoft Azure Marketplace.
+   * [Scalar DB](https://azuremarketplace.microsoft.com/en/marketplace/apps/scalarinc.scalardb)
+   * [Scalar DL](https://azuremarketplace.microsoft.com/en/marketplace/apps/scalarinc.scalardl)
+
+1. Select `Get It Now`.
+
+1. Sign in to the Microsoft Azure Marketplace using your work email address
+   * Please use the work email address that is used as an account of Microsoft Azure.
```
[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/AzureMarketplaceGuide.md]: I see. I agree with we don't use bullet points to separate sentences. Thank you.
I removed unnecessary bullet points in 30061a4fb325d91425fee6c966bcb102cbb142f5.

File: docs/AzureMarketplaceGuide.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/AzureMarketplaceGuide.md]: Ah, sorry. I misunderstood the deference of `*` and `**`...
I fixed to use `**` for emphasizing in ba02dc024ace578412ba85b2e841b2a133066920.

File: docs/AzureMarketplaceGuide.md [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+       * Scalar DL Auditor
+         ```console
+         helm install scalardl-auditor scalar-labs/scalardl-audit -f ./scalardl-auditor-custom-values.yaml
+         ```
+       * Scalar DL Schema Loader
+         ```console
+         helm install schema-loader scalar-labs/schema-loading -f ./schema-loader-custom-values.yaml
+         ```
+
+## Deploy containers on AKS (Azure Kubernetes Service) from your private container registry using Scalar Helm Charts
```
[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/AzureMarketplaceGuide.md]: Yes, that's right. I assume that the AKS is first choice when users subscribe the Azure Marketplace.
I fixed to place the AKS section before the general Kubernetes section in 609c2b9455d97f375547a9e9db38e72d01a99071.

**Comment by kota2and3kan** [comment]
@feeblefakie @ymorimo  
Thank you for your review and correcting english!
I applyed and fixed some pointed out things.
Please take a look again!

**Review by ymorimo — APPROVED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**